website/server/models/group.js

```diff
@@ -1409,29 +1409,12 @@ schema.methods.leave = async function leaveGroup (user, keep = 'keep-all', keepC
   if (group.memberCount <= 1 && group.privacy === 'private') {
     // double check the member count is correct
     // so we don't accidentally delete a group that still has users in it
-    let members;
-    if (group.type === 'guild') {
-      members = await User.find({ guilds: group._id }).select('_id').exec();
-    } else {
-      members = await User.find({ 'party._id': group._id }).select('_id').exec();
-    }
+    this.memberCount = await group.getMemberCount();
```

Comment:
Sorry for the delay @JalanshMunshi , and yes I think that this piece of code will have to be reverted. But the code is correct, the problem we're running into with tests is likely that here https://github.com/HabitRPG/habitica/pull/12308/files#diff-b6f239870392ed0ec2a2b73cbebaa699R1396-R1402 the operation to remove the guild or party from the user is already underway but we're not awaiting for it to finish meaning that when we call 

```
await group.getMemberCount();
```

it could already have finished or not resulting in an incorrect result. For example: It's finished so memberCount doesn't include the user leaving, if there were only 2 members and 1 left doing memberCount - 1 will result in the group being deleted even if there's still a member

So yeah I guess this will have to be reversed, the rest of the PR looks good!

Reply:
Thanks for getting back to me, @paglias !

I have reverted the changes. I guess these will be the final changes for this issue. Please have a final look at the PR and let me know if something is missing.

Reply:
thanks @JalanshMunshi ! The PR looks good, I'll merge it at the end of the week